File: solutions/python/allergies/2/allergies.py
class Allergies:
    ALLERGENS = [
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats"
    ]

    def __init__(self, score):
        self.allergies = [self.ALLERGENS[i] for i in range(len(self.ALLERGENS)) if score & (1 << i)]

    def allergic_to(self, item):
        return item in self.lst

    @property
    def lst(self):
        return self.allergies

File: solutions/r/allergies/2/allergies.R
allergy <- function(num) {
  allergies <- bitwAnd(num, 2^(0:7))
  names(allergies) <- c("eggs", "peanuts", "shellfish", "strawberries", 
                        "tomatoes", "chocolate", "pollen", "cats")
  names(allergies[allergies > 0])
}

allergic_to <- function(allergy_object, allergy) {
  allergy %in% allergy_object
}

list_allergies <- function(allergy_object) {
  allergy_object
}
